src/server/nbs/nbs_player.ts

```diff
@@ -0,0 +1,55 @@
+import * as bindings from 'bindings'
+import { EventEmitter } from 'events'
+
+const NBSPlayerAPI = bindings('nbs_player')
+
+export type NBSPacket = {
+  timestamp: number
+  hash: Buffer
+  payload: Buffer
+}
+
+export class NBSPlayer extends EventEmitter {
```

Comment:
I'd like to avoid extending EventEmitter in favour of delegation and externally typed callback methods

e.g.

```ts
export class NBSPlayer {
  private events: EventEmitter = new EventEmitter();

  constructor(private readonly player: NBSPlayer) {
    this.player.on('packet', this.onNBSPacket);
  }

  // This is the externally typed method of attaching listeners.
  // e.g. new NBSPlayer().onPacket(packet =>{
  // }
  onPacket(callback: (packet: NBSPacket) => void) {
    this.events.on('packet', callback);
  }

  // A private method which handles events from NBSPlayer, which then emits to listeners.
  private onNBSPacket = () => {
    this.events.emit('packet', ...);
  }
}
```